### ISSUE TITLE
[Question] AOIB_workload_generation.py

### ISSUE THREAD

**Issue by Michele-Zhu** [aliyun/aicb#40]
Hello, I'm trying to generate a text workload for further studies on SimAI-ns3 simulation. And I have some questions.
I'm generating workload with the following parameters: 

```
sh ./scripts/megatron_workload_with_aiob.sh -m 22 --world_size 512 --tensor_model_parallel_size 8 /
--pipeline_model_parallel 8 --frame Megatron --global_batch 8192 /
--micro_batch 8 --seq_length 4096 --swiglu --use_flash_attn --aiob_enable
```

```
sh ./scripts/megatron_workload_with_aiob.sh -m 22 --world_size 512 --tensor_model_parallel_size 8 /
--pipeline_model_parallel 8 --frame Megatron --global_batch 8192 /
--micro_batch 32 --seq_length 4096 --swiglu --use_flash_attn --aiob_enable
```

1. The parameter I changed is the number of micro_batches, which affects the number of times the gradient accumulation is performed when all other parameters are fixed. The question arises from the discrepancies between the second line of outputs; in one, I have 1678 description layers, and in the other, I have 430 description layers. The difference in the description is caused by the difference in the execution graph? 

2. Another strange behavior that I don't understand is also the removal of --aoib_enable causes changes that I'm not expecting in the workload description text. With the same configuration of 8 microbatches without aoib, I'm getting a workload description of 1803.
From my understanding, the aoib component is here to measure actual execution times of each layer in the GPU, so it shouldn't affect the workload by itself. The generation code that I have used is this:

```
sh ./scripts/megatron_workload_with_aiob.sh -m 22 --world_size 512 --tensor_model_parallel_size 8 /
--pipeline_model_parallel 8 --frame Megatron --global_batch 8192 /
--micro_batch 32 --seq_length 4096 --swiglu --use_flash_attn 
```

I'm attaching the output files in this order: execution with micro_batch_8, micro_batch 32, micro_batch 8 without aoib

[2025_08_04_gpt22B_tp8_pp8_dp8_mb8.txt](https://github.com/user-attachments/files/21603952/2025_08_04_gpt22B_tp8_pp8_dp8_mb8.txt)

[2025_08_04_gpt22B_tp8_pp8_dp8_mb32.txt](https://github.com/user-attachments/files/21603955/2025_08_04_gpt22B_tp8_pp8_dp8_mb32.txt)

[2025_08_04_gpt22B_tp8_pp8_dp8_mb8_noAIOB.txt](https://github.com/user-attachments/files/21603966/2025_08_04_gpt22B_tp8_pp8_dp8_mb8_noAIOB.txt)

Is my understanding of 1. correct? The behavior in 2. seems strange, I'm misusing the configurations or an unintended behavior? 
P.S. I'm trying to model a execution workload with the following parameters: pp 8, dp 8, tp 8. I'm unsure about the values that I should set for the number of microbatches for an accurate reproduction of a llm workload like gpt22B. Should I just use 1 as microbatch and completely ignore the parameters? From what I understood by decreasing the microbatch you're performing more frequent communication for forward/backward passes, but with smaller communication size. The optimizer.step(), instead is called less frequently thus we are performing less gradient updates.

Execution was performed on NGC container with pytorch and tag 25.06
